mne/viz/epochs.py

```diff
@@ -908,13 +919,20 @@ def _plot_traces(params):
             else:
                 tick_list += [params['ch_names'][ch_idx]]
                 offset = offsets[line_idx]
+
+            if params['inds'][ch_idx] in params['data_picks']:
+                this_decim = params['decim']
+            else:
+                this_decim = 1
             this_data = data[ch_idx]
 
             # subtraction here gets correct orientation for flipped ylim
             ydata = offset - this_data
             xdata = params['times'][:params['duration']]
             num_epochs = np.min([params['n_epochs'], len(epochs.events)])
+
             segments = np.split(np.array((xdata, ydata)).T, num_epochs)
+            segments = np.array(segments)[:, ::this_decim]
```

Comment:
There are some extra memcopy operations here. Can you avoid one of them with `segments = [segment[::this_decim] for segment in segments]` or something similar (not sure the actual sizes)?